File: cosmol_viewer/examples/render_3d_conformer_from_cosmolkit.rs
use cosmol_viewer::{Scene, Viewer, cosmolkit, shapes::Molecule};
use std::path::Path;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let base = cosmolkit::Molecule::from_smiles("Oc1ccccc1-c2ccccc2O")?.sanitize()?;

    let mut params = cosmolkit::EmbedParameters::etkdg_v3();
    params.random_seed = 0xF00D;
    params.num_threads = 1;

    let cosmolkit_mol = base.with_3d_conformer_with_params(params)?;
    let mol = Molecule::from_cosmolkit(&cosmolkit_mol)?
        .centered()
        .enable_outline(0.04);

    let mut scene = Scene::new();
    scene.set_scale(0.9);
    scene.set_camera_view(35.0, 22.0, 0.0, 28.0, [0.0, 0.0, 0.0], 18.0);
    scene.add_shape_with_id("conformer", mol);

    let viewer = Viewer::render(&scene, 800.0, 500.0)?;
    viewer
        .take_screenshot()
        .save(Path::new("cosmolkit_3d_conformer_viewer.png"))?;

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());

    Ok(())
}

File: cosmol_viewer/examples/render_3d_conformer_from_cosmolkit.py
import cosmolkit as ck
from cosmol_viewer import Molecule, Scene, Viewer

base = (
    ck.Molecule.from_smiles("CC(=O)Nc1ccc(O)cc1")
    .with_hydrogens()
    .sanitize()
)

params = ck.EmbedParameters.etkdg_v3()
params.random_seed = 0xF00D
params.num_threads = 1

cosmolkit_mol = base.with_3d_conformer(params)

mol = (
    Molecule.from_cosmolkit(cosmolkit_mol)
    .centered()
    .enable_outline(width=0.04)
)

scene = Scene()
scene.set_scale(1.2)
scene.set_camera_view(azimuth=35, elevation=22, distance=28, fov=18)
scene.add_shape_with_id("conformer", mol)

scene.save_image("render_3d_conformer_from_cosmolkit.png", width=1200, height=900)

viewer = Viewer.render(scene, width=800, height=500)

print("Press Any Key to exit...", end="", flush=True)
try:
    _ = input()
except EOFError:
    pass
